torch/__init__.py

```diff
@@ -1752,6 +1753,109 @@ def registerOp(cls, op_key, full_schema, op_impl, dispatch_key):
             return cls.ops_table[(op_key, dispatch_key)]
 
 
+################################################################################
+# Exposing torch.export() and related support classes
+################################################################################
+
+# TODO(ycao): Move ExportedProgram, Constraint, dynamic_dim etc to torch.compiler namespace
+
+def export(
+    f: Callable,
+    args: Tuple[Any],
+    kwargs: Optional[Dict[str, Any]] = None,
+    *,
+    constraints: Optional[List["torch._dynamo.eval_frame.Constraint"]] = None,
+) -> "torch._export.exported_program.ExportedProgram":  # type: ignore[name-defined]
+    """
+    torch.export() is a one-shot process for capturing a computation graph from
+    a PyTorch program Ahead-of-Time (AOT).
+
+    This function traces either an nn.Module's forward function or a callable
+    containing PyTorch operations and produces an ExportedProgram. The
+    ExportedProgram includes PyTorch operations that perform computations
+    equivalent to those in the given nn.Module or callable.
+
+    In specific terms, torch.export() traces a function `f` by executing it
+    with the provided `args` and `kwargs`. It records the PyTorch operations
+    invoked during execution to produce the ExportedProgram.
+
+    Acceptable types of inputs (for `args` and `kwargs`) and outputs include:
+        - Primitive types, i.e. torch.Tensor, int, float, bool.
+        - (Nested) Data structures comprising of dict, list,
+            tuple, namedtuple and OrderedDict
+        - Dataclass subclass (must be registered with
+            torch._export.utils.register_dataclass_as_pytree_node` first)
+
+    torch.export() specializes the traced program based on the values of
+    inputs that are not torch.Tensors. If you want to preserve dynamic
+    branching behavior in the generated graph, you will need to manually
+    rewrite the source code using control flow operations like
+    `torch.ops.higher_order.cond`.
+
+    While tracing, torch.export() takes note of shape-related assumptions
+    made by the user program and the underlying PyTorch operator kernels.
+    The output ExportedProgram is considered valid only when these
+    assumptions hold true. These assumptions are validated during tracing
```

Comment:
Technically we can use constrain_as_* APIs to check validity of values, but then you probably have to explain the whole thing about min being 2 etc.